packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, CheckboxControl, Modal } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useCallback } from '@wordpress/element';
+
+const EntitiesSavedStatesCheckbox = ( {
+	id,
+	name,
+	changes: { rawRecord },
+	checked,
+	setCheckedById,
+} ) => (
+	<CheckboxControl
+		label={ `${ startCase( name ) }: "${ rawRecord.title || rawRecord.name }"` }
+		checked={ checked }
+		onChange={ useCallback( ( nextChecked ) => setCheckedById( id, nextChecked ), [ id ] ) }
```

Comment:
>It might be expected that setCheckedById should be listed as a dependency

It never changes. It's like an extension of the function returned by `useState` here.

>But to the more general application, I don't know that we really care to use useCallback here:

Yeah, none of the components here are expensive and it's a leaf component.